### DISCUSSION TITLE
Zephyr.bin For STM32MP1 Target is Huge

### DISCUSSION THREAD

**Discussion by georgenyarangi-eaton** [zephyrproject-rtos/zephyr#51970] [Q&A] [answered]
Zephyr.bin built with optimization (-Os) is 256 MB in size. In comparison, the ELF file is only 1.6 MB. Any guidance would be greatly appreciated. Thanks in advance!

![image](https://user-images.githubusercontent.com/101207500/200028604-39b5ea51-114c-43ed-ae9c-57ac0ef9b850.png)

CC: @FRASTM @erwango @arnopo

**Comment by arnopo** [comment, marked as answer]
The reason is that the Cortex-M4 use 2 memory areas which are not contiguous
- the RETREM start at address 0x000000, size 64 KB
- the MCUSRAM start at address 0x10000000 (also aliased at 0x30000000), size 384 kB

The bin format tries to fill with 0 the memory area between the RETRAM and the MCUSRAM.
This format MUST not be used.

**Reply by georgenyarangi-eaton** [reply]
That makes sense! Thank you so much. So, you're recommending using the ELF even in production? The concern there would be the additional information exposed when using an ELF (symbol names etc.)

**Reply by arnopo** [reply]
What about using the zephyr SDK "strip" command to clean debug information?
A BUILD_OUTPUT_STRIPPED config exists that should do the job (never tested)

**Reply by georgenyarangi-eaton** [reply]
OK, I will definitely try the strip command. It seems like our options are limited here so hopefully a stripped ELF works for us.

**Reply by georgenyarangi-eaton** [reply]
I have verified that the stripped ELF works just fine, and it is much much smaller (1.8 MB vs 82 KB).
